frontend/src/AssignmentView/AddAssignmentModal.tsx

```diff
@@ -1,35 +1,76 @@
-import React, { useState } from "react";
+import React, { FormEvent, useContext, useState } from "react";
 import { useNavigate } from "react-router-dom";
 import Container from "react-bootstrap/Container";
 import Form from "react-bootstrap/Form";
 import Button from "react-bootstrap/Button";
 import Alert from "react-bootstrap/Alert";
 import Stack from "react-bootstrap/Stack";
 import Modal from "react-bootstrap/Modal";
+import { userContext } from "../Context/UserContext";
+import { ProfessorAssignmentContext } from "../Context/ProfessorAssignmentContext";
+import { useCookies } from "react-cookie";
 
 // This view is a form for professors to add a assignment
 function AddAssignmentModal() {
+  const user = useContext(userContext);
   const [error, setError] = useState("");
+  const [cookies, setCookies] = useCookies(["jwt"]);
   const nav = useNavigate();
+  const classId = 1;
+  const assignmentId = 1;
 
   const [show, setShow] = useState(false);
+  const [gradingScriptName, setGradingScriptName] = useState("");
+  const [gradingScriptFile, setGradingScriptFile] = useState("");
+  const [dueDate, setDueDate] = useState("");
   const handleClose = () => setShow(false);
   const handleShow = () => setShow(true);
```

Comment:
If these functions are only used once they can be inlined as long as it's readable.

Reply:
only handleshow used 1 time only so I changed that!